fennel/client_tests/test_bounded_idleness.py

```diff
@@ -0,0 +1,201 @@
+import time
+from datetime import datetime
+
+import pandas as pd
+import pytest
+
+import fennel._vendor.requests as requests
+from fennel.datasets import dataset, field
+from fennel.sources import Webhook, source
+from fennel.testing import mock
+
+webhook = Webhook(name="fennel_webhook")
+__owner__ = "saiharsha@fennel.ai"
+
+
+@source(
+    webhook.endpoint("ClicksDS1"),
+    cdc="append",
+    disorder="1d",
+    bounded=True,
+    idleness="10s",
+)
+@dataset
+class BoundedClicksDS:
+    display_id: int = field(key=True)
+    ad_id: int
+    clicked: bool
+    timestamp: datetime = field(timestamp=True)
+
+
+@source(
+    webhook.endpoint("ClicksDS2"), cdc="append", disorder="1d", bounded=False
+)
+@dataset
+class UnBoundedClicksDS:
+    display_id: int = field(key=True)
+    ad_id: int
+    clicked: bool
+    timestamp: datetime = field(timestamp=True)
+
+
+def _log_clicks_data_batch1(client, webhook_endpoint):
```

Comment:
Can you inline them? 

Looking at the test case, I have no idea what is getting logged so it is difficult to understand the expectations. For unit tests, we should definitely prioritize readability over code reusability.

Reply:
Updated the logic to be more readable